src/main.cpp

```diff
@@ -102,14 +102,11 @@ int main(int argc, char* argv[])
     static const std::string executable(XEUS_PYTHON_EXECUTABLE);
     static const std::wstring wexecutable(executable.cbegin(), executable.cend());
 
-// On windows, sys.executable is not properly set with Py_SetProgramName
-// Cf. https://bugs.python.org/issue34725
-// A private undocumented API was added as a workaround in Python 3.7.2.
-#ifdef _WIN32
-    _Py_SetProgramFullPath(const_cast<wchar_t*>(wexecutable.c_str()));
-#else
+    // On windows, sys.executable is not properly set with Py_SetProgramName
+    // Cf. https://bugs.python.org/issue34725
+    // A private undocumented API was added as a workaround in Python 3.7.2.
+    // _Py_SetProgramFullPath(const_cast<wchar_t*>(wexecutable.c_str()));
```

Comment:
Should we call this private API when Python's version is greater than 3.7.1?

Reply:
We could do it separately.

Reply:
Ok!